src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,158 +1,351 @@
+                _NewTabWithProfileHandlers(4);
+                return true;
+            case ShortcutAction::NewTabProfile5:
+                _NewTabWithProfileHandlers(5);
+                return true;
+            case ShortcutAction::NewTabProfile6:
+                _NewTabWithProfileHandlers(6);
+                return true;
+            case ShortcutAction::NewTabProfile7:
+                _NewTabWithProfileHandlers(7);
+                return true;
+            case ShortcutAction::NewTabProfile8:
+                _NewTabWithProfileHandlers(8);
+                return true;
+            case ShortcutAction::NewTabProfile9:
+                _NewTabWithProfileHandlers(9);
+                return true;
+
+            case ShortcutAction::NewWindow:
+                _NewWindowHandlers();
+                return true;
+            case ShortcutAction::CloseWindow:
+                _CloseWindowHandlers();
+                return true;
+            case ShortcutAction::CloseTab:
+                _CloseTabHandlers();
+                return true;
+
+            case ShortcutAction::ScrollUp:
+                _ScrollUpHandlers();
+                return true;
+            case ShortcutAction::ScrollDown:
+                _ScrollDownHandlers();
+                return true;
+            case ShortcutAction::ScrollUpPage:
+                _ScrollUpPageHandlers();
+                return true;
+            case ShortcutAction::ScrollDownPage:
+                _ScrollDownPageHandlers();
+                return true;
+
+            case ShortcutAction::NextTab:
+                _NextTabHandlers();
+                return true;
+            case ShortcutAction::PrevTab:
+                _PrevTabHandlers();
+                return true;
+
+            case ShortcutAction::SwitchToTab0:
+                _SwitchToTabHandlers(0);
+                return true;
+            case ShortcutAction::SwitchToTab1:
+                _SwitchToTabHandlers(1);
+                return true;
+            case ShortcutAction::SwitchToTab2:
+                _SwitchToTabHandlers(2);
+                return true;
+            case ShortcutAction::SwitchToTab3:
+                _SwitchToTabHandlers(3);
+                return true;
+            case ShortcutAction::SwitchToTab4:
+                _SwitchToTabHandlers(4);
+                return true;
+            case ShortcutAction::SwitchToTab5:
+                _SwitchToTabHandlers(5);
+                return true;
+            case ShortcutAction::SwitchToTab6:
+                _SwitchToTabHandlers(6);
+                return true;
+            case ShortcutAction::SwitchToTab7:
+                _SwitchToTabHandlers(7);
+                return true;
+            case ShortcutAction::SwitchToTab8:
+                _SwitchToTabHandlers(8);
+                return true;
+            case ShortcutAction::SwitchToTab9:
+                _SwitchToTabHandlers(9);
+                return true;
+        }

```

Comment:
this is missing a default case.

Reply:
@adiviness Something to consider in the future.   Instead of relying on every switch statement having a default to handle unexpected cases - an alternative is to turn on C4061 & C4062.   This way the compiler will help out and give you an error. (if warnings as errors are on)  This is really helpful when someone adds a enum value.  This will help point out code that should be updated.   Obviously, this isn't something you can just turn on.  It is something that would need to be evaluated.   It is super nice to rely on the compiler for this sort of thing.  Also it will only work for enum and enum classes. Just thought I would throw this out there, I have been using this with clang for awhile.  The msft compiler wasn't always reliable with this warning but it appears that it may have caught up now.

Reply:
I actually believe that we already have those warnings turned on, in `/src/common.build.pre.props` we set the warning level to 4 and have warnings treated as errors enabled. We do disable a few specific warnings that maybe we can eventually revisit (nameless structs is one that I don't think we can get rid of for compatibility but maybe we can narrow the scope of which projects the warning is disabled in).

Reply:
@adiviness It looks like Level 4 warnings are on, but unfortunately C4016 & C4062 are turned off by default so you won't get these warnings.    To turn these on you need to explicitly [turn these on](https://docs.microsoft.com/en-us/cpp/preprocessor/compiler-warnings-that-are-off-by-default?view=vs-2019). (even if you already turned on Level 4)

Reply:
Interesting, I'll make an issue to get these additional warnings turned on, thanks!